### PR TITLE
en, zh: add release notes for v1.2.0-beta.2

### PR DESCRIPTION
**Pull request by csuzhangxc**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-tidb-operator) that's required for repo owners to accept my contribution.

### What is changed, added, or deleted? (Required)

add release notes for v1.2.0-beta.2

### Which TiDB Operator version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [ ] v1.1 (TiDB Operator 1.1 versions)
- [ ] v1.0 (TiDB Operator 1.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: 
- Other reference link(s): 

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: en/releases/release-1.2.0-beta.2.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+- Support deploying DmCluster in a different namespace of a TidbCluster ([#3914](https://github.com/pingcap/tidb-operator/pull/3914), [@csuzhangxc](https://github.com/csuzhangxc))
+- Support installing TiDB Operator with namespace scoped permissions ([#3896](https://github.com/pingcap/tidb-operator/pull/3896), [@csuzhangxc](https://github.com/csuzhangxc))
+
+## Improvements
+
+- Add readiness probe for the TidbMonitor Pod ([#3943](https://github.com/pingcap/tidb-operator/pull/3943), [@mikechengwei](https://github.com/mikechengwei))
+- Optimize TidbMonitor for DmCluster with TLS enabled ([#3942](https://github.com/pingcap/tidb-operator/pull/3942), [@mikechengwei](https://github.com/mikechengwei))
+- TidbMonitor supports not generating Prometheus alert rules ([#3932](https://github.com/pingcap/tidb-operator/pull/3932), [@mikechengwei](https://github.com/mikechengwei))
+- Optimize the `PodsAreChanged` function ([#3901](https://github.com/pingcap/tidb-operator/pull/3901), [@shonge](https://github.com/shonge))
```
Comment (DanielZhangQD) [inline en/releases/release-1.2.0-beta.2.md:29]: This can be removed.

**Review by DanielZhangQD — COMMENTED** [review]
Rest LGTM

File: en/releases/release-1.2.0-beta.2.md (hunk 1 above)
Comment (csuzhangxc) [reply en/releases/release-1.2.0-beta.2.md]: removed in https://github.com/pingcap/docs-tidb-operator/pull/1196/commits/89822699bdc72b4d1f06fdbfab3aad043c65dc8b

**Review by DanielZhangQD — APPROVED** [review]
LGTM

**Comment by DanielZhangQD** [comment]
@qiancai PTAL.

File: en/releases/release-1.2.0-beta.2.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+TiDB Operator version: 1.2.0-beta.2
+
+## Rolling Update Changes
+
+- Upgrading TiDB Operator will cause the recreation of the TidbMonitor Pod due to [#3943](https://github.com/pingcap/tidb-operator/pull/3943)
+- Upgrading TiDB Operator will cause the recreation of the DM-master Pod due to [#3914](https://github.com/pingcap/tidb-operator/pull/3914)
+
+## New Features
+
+- TidbMonitor supports monitoring multiple TidbCluster with TLS enabled ([#3867](https://github.com/pingcap/tidb-operator/pull/3867), [@mikechengwei](https://github.com/mikechengwei))
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:18]:
```suggestion
- TidbMonitor supports monitoring multiple TidbClusters with TLS enabled ([#3867](https://github.com/pingcap/tidb-operator/pull/3867), [@mikechengwei](https://github.com/mikechengwei))
```

File: en/releases/release-1.2.0-beta.2.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+- Support configuring `podSecurityContext` for all TiDB components ([#3909](https://github.com/pingcap/tidb-operator/pull/3909), [@liubog2008](https://github.com/liubog2008))
+- Support configuring `topologySpreadConstraints` for all TiDB components ([#3937](https://github.com/pingcap/tidb-operator/pull/3937), [@liubog2008](https://github.com/liubog2008))
+- Support deploying DmCluster in a different namespace of a TidbCluster ([#3914](https://github.com/pingcap/tidb-operator/pull/3914), [@csuzhangxc](https://github.com/csuzhangxc))
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:21]:
```suggestion
- Support deploying a DmCluster in a different namespace than a TidbCluster ([#3914](https://github.com/pingcap/tidb-operator/pull/3914), [@csuzhangxc](https://github.com/csuzhangxc))
```

File: en/releases/release-1.2.0-beta.2.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+- Support installing TiDB Operator with namespace scoped permissions ([#3896](https://github.com/pingcap/tidb-operator/pull/3896), [@csuzhangxc](https://github.com/csuzhangxc))
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:22]:
```suggestion
- Support installing TiDB Operator with only namespace-scoped permissions ([#3896](https://github.com/pingcap/tidb-operator/pull/3896), [@csuzhangxc](https://github.com/csuzhangxc))
```

File: en/releases/release-1.2.0-beta.2.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+
+## Improvements
+
+- Add readiness probe for the TidbMonitor Pod ([#3943](https://github.com/pingcap/tidb-operator/pull/3943), [@mikechengwei](https://github.com/mikechengwei))
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:26]:
```suggestion
- Add the readiness probe for the TidbMonitor Pod ([#3943](https://github.com/pingcap/tidb-operator/pull/3943), [@mikechengwei](https://github.com/mikechengwei))
```

File: en/releases/release-1.2.0-beta.2.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+- Optimize TidbMonitor for DmCluster with TLS enabled ([#3942](https://github.com/pingcap/tidb-operator/pull/3942), [@mikechengwei](https://github.com/mikechengwei))
+- TidbMonitor supports not generating Prometheus alert rules ([#3932](https://github.com/pingcap/tidb-operator/pull/3932), [@mikechengwei](https://github.com/mikechengwei))
+
+## Bug Fixes
+
+- Fix the issue that TiDB instances are kept at TiDB Dashboard after scaled in ([#3929](https://github.com/pingcap/tidb-operator/pull/3929), [@july2993](https://github.com/july2993))
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:32]:
```suggestion
- Fix the issue that TiDB instances are kept in TiDB Dashboard after being scaled in ([#3929](https://github.com/pingcap/tidb-operator/pull/3929), [@july2993](https://github.com/july2993))
```

File: en/releases/release-1.2.0-beta.2.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+- Fix the issue that useless TidbCluster CR sync caused by the update of `lastHeartbeatTime` in `status.tikv.stores` ([#3886](https://github.com/pingcap/tidb-operator/pull/3886), [@songjiansuper](https://github.com/songjiansuper))
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:33]:
```suggestion
- Fix the useless sync of TidbCluster CR caused by the update of `lastHeartbeatTime` in `status.tikv.stores` ([#3886](https://github.com/pingcap/tidb-operator/pull/3886), [@songjiansuper](https://github.com/songjiansuper))
```

File: en/releases/release-1.2.0-beta.2.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+title: TiDB Operator 1.2.0-beta.2 Release Notes
+---
+
+# TiDB Operator 1.2.0-beta.2 Release Notes
+
+Release date: April 29, 2021
+
+TiDB Operator version: 1.2.0-beta.2
+
+## Rolling Update Changes
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:11]:
```suggestion
## Rolling update changes
```

File: en/releases/release-1.2.0-beta.2.md [hunk 9]
```diff
@@ -0,0 +1,33 @@
...
+
+- Upgrading TiDB Operator will cause the recreation of the TidbMonitor Pod due to [#3943](https://github.com/pingcap/tidb-operator/pull/3943)
+- Upgrading TiDB Operator will cause the recreation of the DM-master Pod due to [#3914](https://github.com/pingcap/tidb-operator/pull/3914)
+
+## New Features
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:16]:
```suggestion
## New features
```

File: en/releases/release-1.2.0-beta.2.md [hunk 10]
```diff
@@ -0,0 +1,33 @@
...
+- Support deploying DmCluster in a different namespace of a TidbCluster ([#3914](https://github.com/pingcap/tidb-operator/pull/3914), [@csuzhangxc](https://github.com/csuzhangxc))
+- Support installing TiDB Operator with namespace scoped permissions ([#3896](https://github.com/pingcap/tidb-operator/pull/3896), [@csuzhangxc](https://github.com/csuzhangxc))
+
+## Improvements
+
+- Add readiness probe for the TidbMonitor Pod ([#3943](https://github.com/pingcap/tidb-operator/pull/3943), [@mikechengwei](https://github.com/mikechengwei))
+- Optimize TidbMonitor for DmCluster with TLS enabled ([#3942](https://github.com/pingcap/tidb-operator/pull/3942), [@mikechengwei](https://github.com/mikechengwei))
+- TidbMonitor supports not generating Prometheus alert rules ([#3932](https://github.com/pingcap/tidb-operator/pull/3932), [@mikechengwei](https://github.com/mikechengwei))
+
+## Bug Fixes
```
Comment (qiancai) [inline en/releases/release-1.2.0-beta.2.md:30]:
```suggestion
## Bug fixes
```

**Pull request merged**